core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -3745,6 +3745,131 @@ LogicalProject(ENAME=[$1])
 ]]>
         </Resource>
     </TestCase>
+    <TestCase name="testConvertMultiFullOuterJoinRule">
+        <Resource name="sql">
+            <![CDATA[select e1.ename from emp e1 full outer join dept d on e1.deptno = d.deptno full outer join emp e2 on d.deptno = e2.deptno]]>
+        </Resource>
+        <Resource name="planBefore">
+            <![CDATA[
+LogicalProject(ENAME=[$1])
+  LogicalJoin(condition=[=($9, $18)], joinType=[full])
+    LogicalJoin(condition=[=($7, $9)], joinType=[full])
+      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+      LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
+    LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+]]>
+        </Resource>
+        <Resource name="planAfter">
+            <![CDATA[
+LogicalProject(ENAME=[$1])
+  MultiJoin(joinFilter=[AND(=($9, $18), =($7, $9))], isFullOuterJoin=[true], joinTypes=[[INNER, INNER, INNER]], outerJoinConditions=[[NULL, NULL, NULL]], projFields=[[ALL, ALL, ALL]])
```

Comment:
Output from the rule does not seem correct, since joinType is registered as INNER and joinFilter has merged all filter conditions for both FOJ.

Reply:
The original rule (without this change) converts full outer joins to multijoin with "INNER" type but with isFullOuterJoin set to true.  From your comment, it looks like this isn't intentional.  Just out of curiosity, why is there a separate isFullOuterJoin flag then?

For example, even without this patch, the following query

`select e1.ename from emp e1 full outer join dept d on e1.deptno = d.deptno`

is converted to 

`LogicalProject(ENAME=[$1])
  MultiJoin(joinFilter=[=($7, $9)], isFullOuterJoin=[true], joinTypes=[[INNER, INNER]], outerJoinConditions=[[NULL, NULL]], projFields=[[ALL, ALL]])
    LogicalTableScan(table=[[CATALOG, SALES, EMP]])
    LogicalTableScan(table=[[CATALOG, SALES, DEPT]])`